File: Go/XLog/level_test.go
// Copyright (c) 2026 FrameworkEase Technologies. All rights reserved.
// Use of this source code is governed by a MIT-style
// license that can be found in the LICENSE file.

package XLog

import (
	"testing"

	"github.com/stretchr/testify/assert"
)

func TestTypes(t *testing.T) {
	tests := []struct {
		name  string
		level Levels
		want  int8
	}{
		{"Unknown", LevelUnknown, 0},
		{"Emergency", LevelEmergency, 1},
		{"Alert", LevelAlert, 2},
		{"Critical", LevelCritical, 3},
		{"Error", LevelError, 4},
		{"Warn", LevelWarn, 5},
		{"Notice", LevelNotice, 6},
		{"Info", LevelInfo, 7},
		{"Debug", LevelDebug, 8},
	}
	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) { assert.Equal(t, tt.want, int8(tt.level)) })
	}
}

func TestLabels(t *testing.T) {
	expected := []string{"[U]", "[M]", "[A]", "[C]", "[E]", "[W]", "[N]", "[I]", "[D]"}

	assert.Equal(t, len(expected), len(Labels))

	for i, want := range expected {
		assert.Equal(t, want, Labels[i])
	}

	levels := []Levels{LevelUnknown, LevelEmergency, LevelAlert, LevelCritical, LevelError, LevelWarn, LevelNotice, LevelInfo, LevelDebug}
	for i, level := range levels {
		assert.Equal(t, expected[i], Labels[level])
	}
}

File: CSharp/tests~/Assets/Tests/Runtime/Scripts/XLog/level_test.cs
// Copyright (c) 2026 FrameworkEase Technologies. All rights reserved.
// Use of this source code is governed by a MIT-style
// license that can be found in the LICENSE file.

using FrameworkEase.Universal.X.Utility;
using NUnit.Framework;

public partial class TestXLog
{
    [Test]
    public void Levels()
    {
        var tests = new[]
        {
            new { name = "Unknown", level = XLog.Levels.Unknown, want = (sbyte)0 },
            new { name = "Emergency", level = XLog.Levels.Emergency, want = (sbyte)1 },
            new { name = "Alert", level = XLog.Levels.Alert, want = (sbyte)2 },
            new { name = "Critical", level = XLog.Levels.Critical, want = (sbyte)3 },
            new { name = "Error", level = XLog.Levels.Error, want = (sbyte)4 },
            new { name = "Warn", level = XLog.Levels.Warn, want = (sbyte)5 },
            new { name = "Notice", level = XLog.Levels.Notice, want = (sbyte)6 },
            new { name = "Info", level = XLog.Levels.Info, want = (sbyte)7 },
            new { name = "Debug", level = XLog.Levels.Debug, want = (sbyte)8 },
        };

        foreach (var tt in tests) Assert.That((sbyte)tt.level, Is.EqualTo(tt.want));
    }

    [Test]
    public void Labels()
    {
        var expected = new[] { "[U]", "[M]", "[A]", "[C]", "[E]", "[W]", "[N]", "[I]", "[D]" };

        Assert.That(XLog.Labels.Length, Is.EqualTo(expected.Length));
        for (var i = 0; i < expected.Length; i++) Assert.That(XLog.Labels[i], Is.EqualTo(expected[i]));

        var levels = new[]
        {
            XLog.Levels.Unknown,
            XLog.Levels.Emergency,
            XLog.Levels.Alert,
            XLog.Levels.Critical,
            XLog.Levels.Error,
            XLog.Levels.Warn,
            XLog.Levels.Notice,
            XLog.Levels.Info,
            XLog.Levels.Debug
        };
        for (var i = 0; i < levels.Length; i++) Assert.That(XLog.Labels[(int)levels[i]], Is.EqualTo(expected[i]));
    }
}
